app/Cmd.hs

```diff
@@ -0,0 +1,71 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Cmd (
+  CmdOpts (CmdOpts),
+  MongoOpts (MongoOpts)
+  ) where
+
+import           Data.Bifunctor
+import           Data.Maybe
+import           Data.Text        hiding (group)
+import           Database.MongoDB hiding (Group, group)
+import           Options
+
+-------------------------------------------------------------------------------------------------------------------------
+-- Global options
+-------------------------------------------------------------------------------------------------------------------------
+data CmdOpts = CmdOpts {
+  mongoOpts :: MongoOpts,
+  port      :: Int
+  } deriving (Eq, Show)
+
+instance Options CmdOpts where
+  defineOptions = pure CmdOpts
+                  <*> defineOptions
+                  <*> simpleOption "port" 3000 "Port on which to listen for incoming HTTP calls."
+
+-------------------------------------------------------------------------------------------------------------------------
+-- MongoDB options
+-------------------------------------------------------------------------------------------------------------------------
+
+data MongoOpts = MongoOpts {
+  mongoHost :: Host,
+  mongoDb   :: Text,
+  mongoCol  :: Text
+  } deriving (Eq, Show)
+
+mongoGroup :: Group
+mongoGroup = group "mongodb" "MongoDB options" "Show MongoDB specific options"
+
+optionType_host :: OptionType Host
+optionType_host = optionType "mongodb host" (host "localhost") parseHost showHostPort
```

Comment:
The mix of camelCase and snake_case is a little odd -- the idiom in Haskell is camelCase, but you do *occasionally* see `snake_case`. I try to avoid it whenever possible.

Reply:
Absolutely - it also comes up as a warning  with my linter - but this is how `Options` defines its its internal option types: `optionType_string`, `optionType_int`... and I thought it'd be better to be consistent for people familiar with the library. Is that not the right call, at least in the Haskell community?

Reply:
That's a pretty reasonable convention to follow. I would ignore the linter in this case.